src/mongod_helpers.py

```diff
@@ -167,6 +165,40 @@ def initialise_replica_set(self, hosts: list) -> None:
         finally:
             client.close()
 
+    def remove_replica(self, remove_replica_ip: str) -> None:
+        """Removes the replica with remove_replica_ip from the MongoDB replica set config.
+
+        Using the current MongoDB configuration from mongod, remove_replica reconfigures the
+        replica set such that it no longer contains the replica with remove_replica_ip.
+        """
+        replica_set_client = self.client()
+        try:
+            # get current configuration and update with correct hosts
+            rs_config = replica_set_client.admin.command("replSetGetConfig")
+            rs_config["config"]["_id"] = self._replica_set_name
+            rs_config["config"]["version"] += 1
+
+            # acquire current members in config
+            current_members = rs_config["config"]["members"]
+            logger.debug("current config for replica set: %s", current_members)
+            new_members = []
+
+            # add members to new config that aren't getting removed
+            for member in current_members:
+                # get member ip without ":PORT"
+                member_ip = member["host"].split(":")[0]
+
+                # if this ip is still being used retain it in new config
+                if not member_ip == remove_replica_ip:
```

Comment:
This order of operations feels a little clunky - could you use != instead?

Reply:
Ah, sorry for some reason I though I couldn't do `!=`